contracts/contracts/TACoApplication.sol

```diff
@@ -228,6 +216,26 @@ contract PREApplication is IApplication, Adjudicator, OwnableUpgradeable {
         __Ownable_init();
     }
 
+    /**
+     * @notice Set contract for multi-chain interactions
+     */
+    function setUpdatableStakeInfo(IUpdatableStakeInfo _updatableStakeInfo) external onlyOwner {
+        require(address(_updatableStakeInfo) != address(updatableStakeInfo), "New address must not be equal to the current one");
```

Comment:
Do we really need this check? What's the worst can happen?

Reply:
considering that is method for DAO - I prefer to more checks than less in any case